### ISSUE TITLE
`--kube-feature-gates ` doesn't work in `kwokctl --runtime=kind` (v0.5.1)

### ISSUE THREAD

**Issue by Huang-Wei** [kubernetes-sigs/kwok#1041]
### How to use it?

- [ ] kwok
- [ ] kwokctl --runtime=docker (default runtime)
- [ ] kwokctl --runtime=binary
- [ ] kwokctl --runtime=nerdctl
- [X] kwokctl --runtime=kind

### What happened?

When running the following command:

```
kwokctl create cluster --runtime kind --kube-feature-gates MatchLabelKeysInPodAffinity=true
```

It reports:

```
ERROR: failed to create cluster: unable to decode config: yaml: unmarshal errors:
  line 86: cannot unmarshal !!seq into map[string]bool
ERROR Execute exit err="failed to start cluster \"kwok-kwok\": cmd wait: kind create cluster --config /Users/weih/.kwok/clusters/kwok/kind.yaml --name kwok-kwok --image docker.io/kindest/node:v1.29.0 --wait 1m: exit status 1"
```

### What did you expect to happen?

--kube-feature-gates works as expected.

### How can we reproduce it (as minimally and precisely as possible)?

See the steps in "What happened" section

### Anything else we need to know?

It seems the bug was introduced back in  #440, and continues to v0.5.1. However, recent refactoring PR #1021 removes the `kind.yaml.tpl` and hence fix this issue.

### Kwok version

<details>

```console
$ kwok --version
kwok version v0.5.1 go1.21.7 (darwin/arm64)
$ kwokctl --version
kwokctl version v0.5.1 go1.21.7 (darwin/arm64)
```

</details>

### OS version

<details>
```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Darwin:
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```
</details>

**Comment by Huang-Wei** [comment]
@wzshiming this can be a hot fix in **v0.5.2** (if it's planned), the solution can be either revert #440, or cherrypick #1021.

**Comment by wzshiming** [comment]
Fixed by #1047

/close
